rslib/src/card_rendering/parser.rs

```diff
@@ -0,0 +1,261 @@
+// Copyright: Ankitects Pty Ltd and contributors
+// License: GNU AGPL, version 3 or later; http://www.gnu.org/licenses/agpl.html
+
+use std::collections::HashMap;
+
+use nom::{
+    branch::alt,
+    bytes::complete::is_not,
+    bytes::complete::tag,
+    character::complete::{anychar, multispace0},
+    combinator::{map, not, recognize, success, value},
+    multi::{many0, many1},
+    sequence::{delimited, pair, preceded, separated_pair, terminated, tuple},
+};
+
+use super::{CardNodes, Node, OtherTag, Tag, TtsTag};
+
+type IResult<'a, O> = nom::IResult<&'a str, O>;
+
+impl<'a> CardNodes<'a> {
+    pub(super) fn parse(mut txt: &'a str) -> Self {
+        let mut nodes = Vec::new();
+        while let Ok((remaining, node)) = node(txt) {
+            txt = remaining;
+            nodes.push(node);
+        }
+
+        Self(nodes)
+    }
+}
+
+impl<'a> Tag<'a> {
+    fn new(name: &'a str, options: Vec<(&'a str, &'a str)>, content: &'a str) -> Self {
+        match name {
+            "tts" => {
```

Comment:
Currently using case-sensitive names.

Reply:
Most things Anki exposes are case sensitive, so that's the consistent approach :-)